test/metabase/lib/expression_test.cljc

```diff
@@ -301,6 +303,24 @@
     (is (= (lib/visible-columns query)
            (lib/expressionable-columns query nil)))))
 
+(deftest ^:parallel expressionable-columns-exclude-expressions-containing-offset
+  (testing "expressionable-columns should filter out expressions which contain :offset"
+    (let [query (-> lib.tu/venues-query
+                    (lib/order-by (meta/field-metadata :venues :id) :asc)
+                    (lib/expression "Offset col"    (lib/offset (meta/field-metadata :venues :price) -1))
+                    (lib/expression "Nested Offset"
+                                    (lib/* 100 (lib/offset (meta/field-metadata :venues :price) -1))))]
+      (testing (lib.util/format "Query =\n%s" (u/pprint-to-str query))
+        (is (=? [{:id (meta/id :venues :id) :name "ID"}
+                 {:id (meta/id :venues :name) :name "NAME"}
+                 {:id (meta/id :venues :category-id) :name "CATEGORY_ID"}
+                 {:id (meta/id :venues :latitude) :name "LATITUDE"}
+                 {:id (meta/id :venues :longitude) :name "LONGITUDE"}
+                 {:id (meta/id :venues :price) :name "PRICE"}
+                 {:id (meta/id :categories :id) :name "ID"}
+                 {:id (meta/id :categories :name) :name "NAME"}]
+                (lib/expressionable-columns query -1 2)))))))
```

Comment:
Would it make sense to add a test case showing that they are available at the next stage? (I know there is a test for returned-columns, but still.)

Reply:
Not a bad idea, and I've added it to my own todo list, but I don't think it's worth another CI run for this PR.